hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/JoinNestedLoopPhysicalRel.java

```diff
@@ -109,6 +115,53 @@ public <V> V accept(CreateDagVisitor<V> visitor) {
         return visitor.onNestedLoopJoin(this);
     }
 
+    /**
+     * Cost calculation of Nested Loop Join relation.
+     * <p>
+     * Nested Loop Join algorithm is a simple join algorithm, where for each left row,
+     * we are traversing the whole right row set. Cost estimation is the following: <ol>
+     * <li> L is a count of left side rows.
+     * <li> R is a count of right side rows.
+     * <li> PD is a produced row count: L * R * (join selectivity).
+     * <li> PR is a processed row count: L * k * R, where k is 1 for non-equi-join,
+     * (join selectivity) ≤ k ≤ 1 for equi-join and 1/R for key lookup.
```

Comment:
where is "(join selectivity) ≤ k ≤ 1 for equi-join" and "1/R for key lookup" implemented? The formula in the code is `double processedRowCount = Math.max(1.0, leftRowCount * rightRowCount);`